### ISSUE TITLE
Provide support to read files from directory like Directory Listing Config

### ISSUE THREAD

**Issue by george-pc** [abstracta/jmeter-java-dsl#247]
We need support to read files from directory. The use case is to read sql files from a directory and execute it iteratively. 
It will be great to make the support such that it provides similar properties to iterate only once or continue on EoF based on a flag.

This feature helps to create data driven testing, where the data files and possibly the expected results are in a directory. We can then create a test plan to read and compare the files based on the files present in the directory.

**Comment by geoe6** [comment]
Thank you for your quick and fast responses.
The sql are not part of the setup/tear down. 
Actually right now i have a JMX test plan where i store the queries in csv file and read then using the csv. This works for small files but when the number of enteries in the csv file (queries) or the  query itself becomes large, then maintaining and ready from csv becomes a bit of challenge. I therefore wanted to see if I can place all the sql files in a directory and then the directory reader read them one by one. This could be further enhanced as per below logic when reading the sql from directory
If i run in functional mode then it should run only once and capture the responses and do response assertion.
If i run it in Non functional mode, it can keep repeating the files in directory.

Attached Test Plan for reference:
[Test_Plan_Scale_UP_down_UTG.jmx.zip](https://github.com/abstracta/jmeter-java-dsl/files/13425153/Test_Plan_Scale_UP_down_UTG.jmx.zip)

**Comment by rabelenda** [comment]
Great, thank you for the detailed example and explanation!

If you would like to try to implement and contribute the `directoryFilesDataSet`, I think it would be a really good contribution. Otherwise you might wait until we can dedicate to it (in a few weeks).

Another potential implementation might be something like:

```
String queryVar = "QUERY";
    String poolName = "pool";
    testPlan(
        // using directory listing plugin (https://github.com/Blazemeter/jmeter-bzm-plugins/blob/master/directory-listing/DirectoryListing.md)
        directoryFilesDataSet(new TestResource("sqls"), queryVar)
            .fileContents(),
        jdbcConnectionPool(poolName, Driver.class, "jdbc:postgresql://localhost/my_db")
            .user("user")
            .password("pass"),
        threadGroup(1, 1,
            jdbcSampler(poolName, "${" + queryVar + "}")
        )
    ).run();
```

This would make resulting test plan simpler but more effort to implement since it requires implementation of new JMeter test element.

Regards
